skippkg-finder.py

```diff
@@ -0,0 +1,333 @@
+                pkg.startswith('sle-') or pkg.startswith('sle_'):
+            return True
+        if 'sles' in pkg or 'sled' in pkg:
+            return True
+        if 'sap-' in pkg or '-sap' in pkg or pkg.startswith('sap'):
+            return True
+        if 'eula' in pkg or 'branding' in pkg:
+            return True
+        if pkg.startswith('sle15') or\
+                pkg.startswith('sles15') or\
+                pkg.startswith('suse-migration') or\
+                pkg.startswith('migrate') or\
+                pkg.startswith('kernel-livepatch') or\
+                pkg.startswith('patterns') or\
+                pkg.startswith('supportutils-plugin') or\
+                pkg.startswith('lifecycle-data-sle') or\
+                pkg.startswith('sca-patterns') or\
+                pkg.startswith('susemanager-') or\
+                pkg.startswith('kiwi-template') or\
+                pkg.startswith('desktop-data'):
+            return True
+        if pkg.endswith('bootstrap') or pkg.endswith('-caasp') or\
+                pkg.endswith('-sle'):
+            return True
+        if pkg == 'suse-build-key' or pkg == 'suse-hpc' or\
+                pkg == 'llvm' or\
+                pkg == 'txt2tags' or\
+                pkg == 'zypper-search-packages-plugin' or\
+                pkg == 'python-pymemcache' or\
+                pkg == 'python-ibus':
+            return True
+        return False
+
+    def get_packagelist(self, project, by_project=True):
+        """
+        Return the list of package's info of a project.
+        If the latest package is from an incident then returns incident
+        package.
+        """
+
+        pkglist = {}
+        packageinfo = {}
+        query = {'expand': 1}
+        root = ET.parse(http_GET(makeurl(self.apiurl, ['source', project],
+                                 query=query))).getroot()
+        for i in root.findall('entry'):
+            pkgname = i.get('name')
+            orig_project = i.get('originproject')
+            is_incidentpkg = False
+            if pkgname.startswith('000') or\
+                    pkgname.startswith('_') or\
+                    pkgname.startswith('patchinfo.') or\
+                    pkgname.startswith('skelcd-') or\
+                    pkgname.startswith('installation-images') or\
+                    pkgname.endswith('-mini'):
+                continue
+            # ugly hack for go1.x incidents as the name would be go1.x.xxx
+            if '.' in pkgname and re.match(r'[0-9]+$', pkgname.split('.')[-1]) and \
+                    orig_project.startswith('SUSE:') and orig_project.endswith(':Update'):
+                is_incidentpkg = True
+                if pkgname.startswith('go1') or pkgname.startswith('bazel0') or \
+                        pkgname.startswith('dotnet') or pkgname.startswith('ruby2'):
+                    if not (pkgname.count('.') > 1):
+                        is_incidentpkg = False
```

Comment:
this can be error prone. We are facing something similar in https://github.com/openSUSE/openSUSE-release-tools/pull/2623 and use releasename - https://github.com/openSUSE/openSUSE-release-tools/pull/2623/files#diff-c1d94e5890752eab31a911fc75e1bcada9d337a7315c97eb8fe8c7046e9f7d30R85

Reply:
I know, but the reason is I don't want to file an GET event on every package including every incidents, otherwise the script can not finish in 1 day or 2 even, suppose that all of packages from SLE15:GA, SLE15:Update, SP1:GA, SP1:Update ... SP4:GA + Backports + Leap... code looks ugly indeed for those special case like `go1.11.xxxxx` or others, but it saves time.

Reply:
@imobachgs I think working on making it more efficient, so hopefully in future we will have faster detection of such  cases.

Reply:
to me, if `/source/openSUSE:Leap:15.4?expand=1` can have 1) a `originpacakge` attrib to each entry, and 2) a `releasename` attrib to each entry, then my code can be simplified a lot :)

maybe like

    <entry name="yast2" originproject="SUSE:SLE-15-SP3:Update" originpackage="yast2.19962"/>
    <entry name="yast2.19962" originproject="SUSE:SLE-15-SP3:Update" originpackage="yast2.19962" releasename="yast2"/>

Reply:
yeah, for us too. Maybe we can ask build service guys what they think about it

Reply:
Using the `/source/openSUSE:Leap:15.4?view=info` you can have the origin, but the `releasename` is still missing:

```xml
<sourceinfo package="yast2" rev="2" vrev="3.9" srcmd5="62841f713933618908b699c060d1d747" lsrcmd5="3e4e02a6e681a21402659868e83b6c2a" verifymd5="7189b727ca2be435c4a45fcc4241e7dc">
<filename>yast2.spec</filename>
<originproject>SUSE:SLE-15-SP3:Update</originproject>
<linked project="openSUSE:Backports:SLE-15-SP4" package="yast2"/>
<linked project="SUSE:SLE-15-SP4:GA" package="yast2"/>
<linked project="SUSE:SLE-15-SP3:Update" package="yast2"/>
<linked project="openSUSE:Leap:15.4" package="yast2.19962"/>
<linked project="openSUSE:Backports:SLE-15-SP4" package="yast2.19962"/>
<linked project="SUSE:SLE-15-SP4:GA" package="yast2.19962"/>
<linked project="SUSE:SLE-15-SP3:Update" package="yast2.19962"/>
</sourceinfo>
```

And, as @nilxam commented, an additional GET is not an option. I will ask the build service team for some help.

Reply:
the url `/source/openSUSE:Leap:15.4?expand=1` in `skippkg_finder` gives `originproject` already, just need to find the proper linked incident package in case it pointing to an incident, same for `?view=info` though